scripts_python/create_hdf5_dataset.py

```diff
@@ -0,0 +1,262 @@
+    p.add_argument('training_subjs',
+                   help="txt file containing the list of subjects ids to use "
+                        "for training.")
+    p.add_argument('validation_subjs',
+                   help="txt file containing the list of subjects ids to use "
+                        "for validation.")
+    p.add_argument('--step_size', type=float,
+                   help="Common step size to resample the data. Must be in "
+                        "the same space as the streamlines.\nIf none is "
+                        "given, we will compress the streamlines instead of "
+                        "resampling.")
+    p.add_argument('--name',
+                   help="Dataset name [Default uses date and time of "
+                        "processing].")
+    p.add_argument('--bundles', nargs='+',
+                   help="Bundles to concatenate as streamlines in the hdf5. "
+                        "The bundles names will be guessed from the "
+                        "filenames.\nMust be names of files that are present "
+                        "in each subject's 'bundles' folder in dwi_ml_ready.\n"
+                        "If none is given, we will take all the files in the "
+                        "'bundles' folder.")
+    p.add_argument('--mask',
+                   help="Mask defining the voxels used for data "
+                        "standardization.\nIf none is given, all non-zero "
+                        "voxels will be used. Should be the name of "
+                        "a file inside dwi_ml_ready/{subj_id}.")
+    p.add_argument('--space', type=str, default=Space.VOX,
+                   choices=['Space.RASMM', 'Space.VOX', 'Space.VOXMM'],
+                   help="Default space to bring all the stateful tractograms. "
+                        "All other measures (ex, step_size) should be "
+                        "provided in that space.")
+    p.add_argument('--logging',
+                   choices=['error', 'warning', 'info', 'debug'],
+                   default='warning',
+                   help="Choose logging level. [warning]")
+    p.add_argument('--save_intermediate', action="store_true",
+                   help="If set, save intermediate processing files for "
+                        "each subject inside")
+    p.add_argument('-f', '--force', action='store_true',
+                   help="If set, overwrite existing hdf5 folder.")
+
+    arguments = p.parse_args()
+
+    return arguments
+
+
+def main():
+    """Parse arguments, generate hdf5 dataset and save it on disk."""
+    args = _parse_args()
+
+    # Initialize logger
+    logging.basicConfig(level=str(args.logging).upper())
+
+    # Verify that dwi_ml_ready folder is found
+    dwi_ml_ready_folder = Path(args.database_folder + '/dwi_ml_ready')
+    if not dwi_ml_ready_folder.is_dir():
+        raise ValueError('The dwi_ml_ready folder was not found: '
```

Comment:
Change for "parser.error" ?

Reply:
I don't know the standard here. If you think so. Others agree?

Reply:
I dislike answering a question with another one, but `ArgumentError` or `ArgumentTypeError` could also be considered. Not sure about the downsides/adequacy of these possibilities. But using Antoine's proposal for all parser-related errors might be a good default choice.